src/util/preconditions.js

```diff
@@ -1,34 +1,30 @@
-'use strict';
-
-var errors = require('../errors');
-var _ = require('lodash');
+const _ = require('lodash');
+const buffer = require('buffer');
+const errors = require('../errors');
 
 module.exports = {
-  checkState: function(condition, message) {
+  checkState(condition, message) {
     if (!condition) {
       throw new errors.InvalidState(message);
     }
   },
-  checkArgument: function(condition, argumentName, message, docsPath) {
+  checkArgument(condition, argumentName, message, docsPath) {
     if (!condition) {
       throw new errors.InvalidArgument(argumentName, message, docsPath);
     }
   },
-  checkArgumentType: function(argument, type, argumentName) {
+  checkArgumentType(argument, type, argumentName) {
     argumentName = argumentName || '(unknown name)';
     if (_.isString(type)) {
       if (type === 'Buffer') {
-        var buffer = require('buffer'); // './buffer' fails on cordova & RN
         if (!buffer.Buffer.isBuffer(argument)) {
           throw new errors.InvalidArgumentType(argument, type, argumentName);
         }
-      } else if (typeof argument !== type) {
+      } else if (typeof argument !== type) { // eslint-disable-line valid-typeof
```

Comment:
We should probaby try to refactor this? It is used allo over the place so it will break everything.

Reply:
Yeah, maybe we can create a new pull request so that we fix all instances of this issue at once. We can talk on Monday

Reply:
Follow-up #70